python/ray/_private/test_utils.py

```diff
@@ -952,26 +955,40 @@ async def run(self):
             while self.is_running:
                 node_to_kill_ip = None
                 node_to_kill_port = None
-                nodes = ray.nodes()
-                alive_nodes = self._get_alive_nodes(nodes)
-                for node in nodes:
-                    node_id = node["NodeID"]
-                    # make sure at least 1 worker node is alive.
-                    if (node["Alive"] and node_id != self.head_node_id
-                            and node_id not in self.killed_nodes
-                            and alive_nodes > 2):
-                        node_to_kill_ip = node["NodeManagerAddress"]
-                        node_to_kill_port = node["NodeManagerPort"]
-                        break
+                while node_to_kill_port is None and self.is_running:
+                    nodes = ray.nodes()
+                    alive_nodes = self._get_alive_nodes(nodes)
+                    for node in nodes:
+                        node_id = node["NodeID"]
+                        # make sure at least 1 worker node is alive.
+                        if (node["Alive"] and node_id != self.head_node_id
+                                and node_id not in self.killed_nodes
+                                and alive_nodes > 2):
+                            node_to_kill_ip = node["NodeManagerAddress"]
+                            node_to_kill_port = node["NodeManagerPort"]
+                            break
+                    # Give the cluster some time to start.
+                    await asyncio.sleep(0.1)
+
+                if not self.is_running:
+                    break
+
+                sleep_interval = np.random.rand() * self.node_kill_interval_s
```

Comment:
Is this for adding undeterministic kill interval every time?

Reply:
Yep.